components/board/server.go

```diff
@@ -144,11 +144,14 @@ func (s *serviceServer) ReadAnalogReader(
 		return nil, err
 	}
 
-	val, err := theReader.Read(ctx, req.Extra.AsMap())
+	val, analogRange, err := theReader.Read(ctx, req.Extra.AsMap())
 	if err != nil {
 		return nil, err
 	}
-	return &pb.ReadAnalogReaderResponse{Value: int32(val)}, nil
+	return &pb.ReadAnalogReaderResponse{
+		Value:    int32(val),
+		MinRange: analogRange.Min, MaxRange: analogRange.Max, StepSize: analogRange.StepSize,
```

Comment:
Mild preference to put each of these on its own line. but I don't feel strongly.